integration_tests/src/main/python/array_test.py

```diff
@@ -126,13 +126,12 @@ def test_array_item_with_strict_index(strict_index_enabled, index):
             test_df,
             conf=test_conf)
 
-
 # No need to test this for multiple data types for array. Only one is enough, but with two kinds of invalid index.
-@pytest.mark.skipif(not is_before_spark_330(),
-                    reason="Only in Spark [3.1.1, 3.3.0) with ANSI mode, it throws exceptions for invalid index")
+@pytest.mark.skipif(not (is_before_spark_330() or is_spark_340_or_later()),
+                    reason="In Spark [3.1.1, 3.3.0) U [3.4.0, +inf)  with ANSI mode, it throws exceptions for invalid index")
```

Comment:
This is very odd, since this is essentially saying that on Spark 3.3.x it does not throw an exception for an invalid index which I find very hard to believe.  From my testing, this test passes on Spark 3.3.0 if we change the `is_spark_340_or_later()` to `is_spark330_or_later()`, which makes me think this test should not be skipped for any Spark versions.